src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -197,16 +201,149 @@ class SubcategorySelectionPools<SubgraphPool> {
   constructor(
     public pools: SubgraphPool[],
     public readonly poolsNeeded: number
-  ) {}
+  ) {
+  }
 
   public hasEnoughPools(): boolean {
     return this.pools.length >= this.poolsNeeded;
   }
 }
 
+export type CrossedLiquidityCandidatePools = {
+  v2Pools: V2SubgraphPool[];
+  v3Pools: V3SubgraphPool[];
+}
+
+export async function getMixedCrossLiquidityCandidatePools({
+  tokenIn,
+  tokenOut,
+  routingConfig,
+  v2SubgraphProvider,
+  v3SubgraphProvider,
+  v2Candidates,
+  v3Candidates,
+}: MixedCrossLiquidityCandidatePoolsParams): Promise<CrossedLiquidityCandidatePools> {
+  const blockNumber = routingConfig.blockNumber;
+
+  const v2Pools = (await v2SubgraphProvider.getPools(tokenIn, tokenOut, {
+    blockNumber,
+  })).sort((a, b) => b.reserve - a.reserve);
+  const v3Pools = (await v3SubgraphProvider.getPools(tokenIn, tokenOut, {
+    blockNumber,
+  })).sort((a, b) => b.tvlUSD - a.tvlUSD);
+
+  for (const pool of v2Pools) {
+    pool.token0.id = pool.token0.id.toLowerCase();
```

Comment:
why do we have to do this? shouldnt it always be the same?

Reply:
Removed this code now, yeah its unnecessary